### ISSUE TITLE
Failure during data fetching

### ISSUE THREAD

**Issue by nitrocode** [jackd248/temps#49]
I have a key from openweather and tested it with curl to make sure it works. I enter the key into temps.app and every time I run it, it responds with

> Failure during data fetching

Is there a log file or a way to get more information on why / how it's failing?

**Comment by musicpro** [comment]
I guess it's cause of the API change by open weather map. They want you to pay for the daily forecasts.
I created a fork of this project and fixed the problem, at least I hope so ;)
You can try it if you want.
https://github.com/musicpro/temps

**Comment by GrigorijST** [comment]
> I guess it's cause of the API change by open weather map. They want you to pay for the daily forecasts.
> I created a fork of this project and fixed the problem, at least I hope so ;)
> You can try it if you want.
> https://github.com/musicpro/temps

Thanks. It works awesome.

**Comment by axelv73** [comment]
> I guess it's cause of the API change by open weather map. They want you to pay for the daily forecasts.
> I created a fork of this project and fixed the problem, at least I hope so ;)
> You can try it if you want.
> https://github.com/musicpro/temps

Hi! How can I use the fork?? I don't understand.... Thanks

**Comment by musicpro** [comment]
Well the easiest way will be to clone or download the repo (my fork) and build it for your own. It's done like the README says. 
If you need more help just ask. I even could provide a installer you can download but I did not uploaded one yet. :)

**Comment by axelv73** [comment]
> Well the easiest way will be to clone or download the repo (my fork) and build it for your own. It's done like the README says.
> If you need more help just ask. I even could provide a installer you can download but I did not uploaded one yet. :)

The installer would be great also because I did not understand how to do ...

**Comment by musicpro** [comment]
@axelv73 So I just uploaded some different versions as releases. So just pick the one you need, unzip it and start the app. It's not a installer more like ready to go :)

**Comment by axelv73** [comment]
@musicpro ok thanks!!! It works fine!
Only one things...the location sees me in the near city and not in mine. :(

**Comment by musicpro** [comment]
Good to hear that it is working :)
I am sorry with the localization, but there is not much I can do. We have to rely on the people who created the database for the IP Localization to get a higher accuracy. Maybe I will find a different option to localize. But till then you can use the field "City" under options to search for your city.

**Comment by helptiger** [comment]
@musicpro I'm using your version. Do you have an idea why it shows "Invalid date"?

![Bildschirmfoto 2019-05-07 um 10 11 46](https://user-images.githubusercontent.com/10258902/57283995-9e114e00-70b0-11e9-8b16-6a0b3db82edb.png)

**Comment by musicpro** [comment]
Yeah, I noticed this too. Reason was the google api. So I fixed it and pushed it. If you need to download an executable file you can find new builds under the release tap. 
Hope it's working for you as well. 
Ah, and maybe you have to add a the API Key for openweathermap again.

**Comment by l1t1s** [comment]
an installer would be great, because i also don't understant how to buil my own one

**Comment by musicpro** [comment]
@l1t1s Hej, just check the release tab on GitHub. You can download the program there. Actually there is no installer needed to use the app.

**Comment by l1t1s** [comment]
great. thousand thanks

**Comment by LaP0573** [comment]
@musicpro Awesome work and fix ! Quick question, is there a way to exit from the detailed day's forecast ? Like, I clicked on a day to see the temperature graph, but I have no way to exit that graph (and display the 4 days forecast again). It stays there, even if I go in the options or close/reopen the Temp window.

**Comment by musicpro** [comment]
@LaP0573 Sorry for the late reply and thanks for using my fix!
So if I am not wrong just click the graph again. For me it is working like that. If not i will check the code again :)

**Comment by happydemic** [comment]
@musicpro Thanks a million for patching this beloved app.

For me the ['invalid date'](https://github.com/jackd248/temps/issues/49#issuecomment-489981024) error has come back again. I'm running your latest patched release of May 2019 ([v.0.7.3](https://github.com/musicpro/temps/releases/tag/v0.7.3)). 

Any chance you might be able to fix this again? I would have opened this as an issue in your fork, but I think issues are disabled over there.

**Comment by musicpro** [comment]
@happydemic Hej, glad to hear that you like it :)

So the reason for this is again the change of an API. I notice that there were a lot of request to the TimeZone API. So google decided to start charging for this and I ended up in a kinda huge bill I can't pay. Luckily I was still in the promo time and did not have to pay this one but every upcoming. This is why i had to deactivate it. 
BUT  I am working on a fix for this. So if you can wait a little bit there is an update coming. 
Maybe you or all of you can tell me how much they need the timezone stuff. Would it be enough to get the current time from you computer? At least this is my play to do now. 

Also I have to rethink the geolocation think again I guess. Caused by maybe paying for the API usage in future. IF you have suggestions feel free :)

BTW I activated the issue tab on my fork, hope this will work.

**Comment by happydemic** [comment]
@musicpro Eeek! Great news you don't have to pay.

I have created this as an [issue](https://github.com/musicpro/temps/issues/1) on your fork.
